### ISSUE TITLE
Choosing "Future benefit" invalidates the subsection with no reason why

### ISSUE THREAD

**Issue by joshmfranklin** [18F/e-QIP-prototype#1525]
## Where are you in the form?

**Section:**
Foreign association > Foreign activities > Foreign benefits

**Question:**
Provide the frequency of the benefit

**Field:**

**Component (for example address block):**
Radio button group

## What did you see?

- [x] Validation error
- [ ] Data not saving
- [ ] Language or content issue
- [ ] Other (explain below)

Steps to reproduce: 
1. Answer yes to receiving a foreign benefit
2. fill out all fields and choose `future benefit` for the question `Provide the frequency of the benefit`
3. See that all fields are filled out and no green circle appears in the sidenav
4. Go back and change the answer to `provide the type of benefit` to any other option and complete the page. 
5. see that the green dot appears in the sidenav

tldr: The green dot doesn't appear in the sidenav if `Future benefit` is chosen, even when all fields are marked as valid and complete.

## What did you expect to see?

I expected the green dot to appear if the page was complete and valid.

## Is this error limited to just this field or others in the form?

- [x] Only relevant to just this question
- [ ] Cross-form issue
- [ ] I don't know

## What operating system and browser/version are you using?

**Operating system**
- [ ] Windows
- [x] MacOS

**Browser**
- [x] Chrome
- [x] Firefox
- [ ] IE
- [ ] Edge
- [ ] Other

**Version:**

**Comment by michaelccata** [comment]
Tested, reproduced and this also applies to `One time benefit`  the other two `Continuing benefit` and `Other` validate as expected.

**Comment by christopherhuii** [comment]
It looks like there's something bigger than this and it would help if someone can verify.

When `Future benefit` is selected, the response in `Provide the date the benefit will begin` does not save. Can someone confirm?

It looks like I wrote some of this code when I first started on this project and there's a discrepancy between `Begin` and `Began`. `One time benefit` and `Continuing benefit` both use the key `Began`, whereas `Future benefit` uses `Begin` for grammar.

This looks and smells like a backend change. I also want to make sure the XML is correct for this, but first we should confirm that the data is truly not saving.

**Comment by christopherhuii** [comment]
This might be an issue, but I'm not 100% sure.

1. Select `Onetime benefit`
2. Fill out all relevant fields
3. Save
4. Choose a different benefit frequency
5. Fill out all relevant fields
6. Save
7. Click `Onetime benefit` and notice all previous data is still persisted.

**Comment by christopherhuii** [comment]
☝️@michaelccata @joshmfranklin ☝️

**Comment by joshmfranklin** [comment]
@christopherhuii yeah I have experienced the same thing. I think since some fields are used by multiple of those branches then they do not clear the data when switching between them. This seems like an edge case to me that we might not need to worry about too much. @michaelccata what are your thoughts?

**Comment by michaelccata** [comment]
Yeah, I guess that depends on whether that data is also sent downstream...which...I guess it probably is if it's getting stored. I'd prefer to fix the persistence issue and get the branch-clearing fixed as well.

**Comment by ryanhofdotgov** [comment]
Branch state not clearing data one path is chosen instead of another has been a significant source of XML generation defects – the logic will key off whether or not a particular property is set (e.g., `state: "NY"`) and you get situations where you have to work-around bogus data being present. For example, a value for state when the location is international.
